src/settings/toml/deploy_config.rs

```diff
@@ -109,37 +116,31 @@ pub struct Zoned {
 }
 
 #[derive(Debug)]
-pub struct RouteConfig {
+pub struct InvocationConfig {
     pub workers_dev: Option<bool>,
     pub route: Option<String>,
     pub routes: Option<Vec<String>>,
     pub zone_id: Option<String>,
     pub account_id: Option<String>,
 }
 
-impl RouteConfig {
-    fn is_valid(&self) -> bool {
-        self.workers_dev_false_by_itself() || self.has_conflicting_targets()
-    }
-
+impl InvocationConfig {
     fn has_conflicting_targets(&self) -> bool {
-        if self.is_zoneless() {
-            self.has_routes_defined()
-        } else if let Some(routes) = &self.routes {
-            !routes.is_empty() && self.route.is_some()
-        } else {
-            false
-        }
+        [
+            self.is_zoneless() && self.is_zoned(),
```

Comment:
this case confused me a bit, since `is_zoneless && has_routes_defined` seemed like a less strict way of specifying `is_zoneless && is_zoned`.

I found that in `settings/toml/environments.rs`, the top level zone id was being copied into an environments InvocationConfig even if the environment was zoneless. I changed this behavior since it didn't make sense, but if I'm missing some reason that it exists, I can revert and add a comment explaining why.